webapp/Connector/src/grid/Panel.js

```diff
@@ -129,19 +129,23 @@ Ext.define('Connector.grid.Panel', {
             {
                 groups.push({
                     text: key,
-                    columns: value.sort(function(a, b)
-                            {
-                                var ah = isMeasure ? a.measure.label.toLowerCase() : a.header.toLowerCase(),
-                                        bh = isMeasure ? b.measure.label.toLowerCase() : b.header.toLowerCase();
-
-                                if (ah.toLowerCase() === Connector.studyContext.subjectColumn.toLowerCase() || ah.toLowerCase() === Connector.studyContext.subjectLabel.toLowerCase())
-                                    return -1;
-                                else if (bh.toLowerCase() === Connector.studyContext.subjectColumn.toLowerCase() || bh.toLowerCase() === Connector.studyContext.subjectColumn.toLowerCase())
-                                    return 1;
-
-                                // sort columns alphabetically by title
-                                return ah == bh ? 0 : (ah > bh ? 1 : -1);
-                            })
+                    columns: value.sort(function(a, b) {
+                        if (isMeasure && a.measure.isVirusMetadataField && b.measure.isVirusMetadataField){
```

Comment:
This sort function doesn't seem quite right to me, though it probably does work for the specific scenarios you need, for other coincidental reasons. Consider the following 3 objects:
A: {isVirusMetadataField: true, sortOrder:100, columnName: colA}
B: {isVirusMetadataField: true, sortOrder:1, columnName: colZ}
C: {z, isVirusMetadataField: false, columnName: colC}

 compare A and B returns 1 (A > B), compare A and C returns -1 (A < C), compare B and C returns 1 (B > C). So the sort function is not transitive.

Reply:
On manual testing I find that indeed this sort function doesn't work and I'm not getting ordered columns on grid as expected, probably due to its broken transitivity.

Reply:
This should be now fixed, please review and test again. Thanks, Xing!